meetings/working-groups/roles/extensions-wg-2023-06-07.md

```diff
@@ -0,0 +1,66 @@
+# Extensions 2023-06-07
+
+## Reviewing the "invocation expression", "member access" and "extension invocation" sections
+
+We reviewed the proposals in PR https://github.com/dotnet/csharplang/pull/7179 and discussed some open issues.  
+
+### Invoking members from extended type
+
+The first issue is that we want to access members from the extended type on a type or value of an extension type.
+
+```csharp
+extension E for C
+{
+  void M()
+  {
+    this.MethodFromUnderlying(); // problem
+    MethodFromUnderlying(); // problem
+  }
+}
+```
+
+The current proposal had a bullet in "member access" 
+which would make member access scenarios work for non-invocations, 
+but that left invocations and simple name scenarios unsupported.
+
+Decision: we want those scenarios to work and we can achieve that 
+by modifying "member lookup" to also find members from the underlying type.
+The "member lookup" section already handles lookups 
+
+### Pattern-based method binding
+
+The second issue is that the proposed rules raise a question about how pattern-based invocations,
+such as the one involved in deconstruction.
+
+For instance, should the following work?
+```csharp
+implicit extension E for C
+{
+   public dynamic Deconstruct { get { ... } } 
+}
+
+class C
+{
+   void M()
+   {
+     (int i, int j) = this;
+   }
+}
+```
+
+Decision: given that an instance `Deconstruct` property would not participate in deconstruction,
+we don't want an extension `Deconstruct` property to participate either.
```

Comment:
I would prefer to state the more general "if the scenario allowed instance properties today, then extension properties should work, if not, then not" rule. The way this is written implies that rule, but doesn't explicitly state it.